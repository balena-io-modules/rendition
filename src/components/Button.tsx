import * as React from 'react';
import { ButtonProps } from 'rendition';
import styled, { StyledFunction, withTheme } from 'styled-components';
import asRendition from '../asRendition';
import {
	bold,
	darken,
	getColor,
	getColoringType,
	lighten,
	normal,
	px,
} from '../utils';

interface ThemedButtonProps extends ButtonProps {
	theme: Theme;
}

const squareWidth = (val: number): number => val / 9 * 10;

const minWidth = (props: ThemedButtonProps) => {
	if (props.w == null && !props.square) {
		return 'auto';
	}
	if (props.square) {
		return props.emphasized
			? squareWidth(props.theme.space[5])
			: squareWidth(props.theme.space[4]);
	}

	return props.w;
};

const horizontalPadding = (props: ButtonProps) => {
	if (props.w == null && !props.square) {
		return props.emphasized ? 50 : 30;
	}
	if (props.square || props.w) {
		return 0;
	}

	return props.emphasized ? 50 : 30;
};

const ButtonIcon = styled.span`
	margin-right: ${props => px(props.theme.space[2])};
	font-size: 0.875em;
`;

const Button = (styled.button as StyledFunction<ThemedButtonProps>)`
	padding: 0 ${props => px(horizontalPadding(props))};
	font-family: inherit;
	display: inline-flex;
	align-items: center;
	justify-content: center;
	font-weight: ${props => normal(props)};
	border-radius: ${props => px(props.theme.radius)};
	appearance: none;
	text-decoration: none;
	border: 0;
	margin: 0;
	min-width: ${props => px(minWidth(props))};
	vertical-align: ${props => props.vAlign || 'top'};
	font-size: inherit;
	line-height: 1.1;
	text-align: center;
	cursor: pointer;
	background: ${props => props.bg || getColor(props, 'bg', 'main')};
	color: ${props => props.color || '#fff'};
	height: ${(props: ThemedButtonProps) =>
		px(props.emphasized ? props.theme.space[5] : props.theme.space[4])};
	transition-property: color, background, border-color;
	transition-duration: .1s;
	transition-timing-function: ease-in;
	outline: none;

	&:hover {
		color: ${props => props.color || '#fff'};
			background-color: ${props =>
				props.bg
					? lighten(props.bg as string)
					: lighten(getColor(props, 'bg', 'main'))}
		}
	}
	&:active {
		color: ${props => props.color || '#fff'};
		background-color: ${props =>
			props.bg
				? darken(props.bg as string)
				: darken(getColor(props, 'bg', 'main'))}
		}

	&:disabled {
		opacity: 0.65;
		cursor: not-allowed;
	}
`;

const Outline = Button.extend`
	color: ${props =>
		getColor(props, 'color', 'main') || props.theme.colors.text.main};
	background: ${props => props.color || 'none'};
	border: 1px solid ${props =>
		props.bg ? props.bg : getColor(props, 'bg', 'main')}
	}
`;

const Plaintext = Button.extend`
	padding-left: 0;
	padding-right: 0;
	height: auto;
	border: 0;
	border-radius: 0;
	color: ${props =>
		getColor(props, 'color', 'main') || props.theme.colors.text.main};
	background: ${props => props.color || 'none'};
	font-weight: ${props => normal(props)};

	&:hover,
	&:focus,
	&:active {
		background: none;
		color: ${props =>
			getColor(props, 'color', 'dark') || props.theme.colors.text.main};
	}
`;

const Underline = Plaintext.extend`
	padding-bottom: 2px;
	border-bottom: 1px solid;
	font-weight: ${props => bold(props)};

	&:hover,
	&:focus,
	&:active {
		color: ${props =>
			getColor(props, 'color', 'main') || props.theme.colors.text.main};
		box-shadow: 0px -1px 0 0px inset;
	}
`;

const DefaultButton = Button.extend`
	background: #fff;
	color: ${props => props.theme.colors.text.main};
	border: 1px solid ${props => props.theme.colors.gray.main};
	&:hover {
		background: #fff;
		color: ${props => props.theme.colors.text.main};
		border-color: ${props => props.theme.colors.text.main};
	}
	&:active {
		color: #fff;
		background-color: ${props => props.theme.colors.text.main};
	}
`;

export default withTheme(
	asRendition(
		({
			outline,
			underline,
			plaintext,
			children,
			iconElement,
			...props
		}: ThemedButtonProps) => {
			if (plaintext) {
				return (
					<Plaintext {...props}>
						{iconElement && <ButtonIcon>{iconElement}</ButtonIcon>}
						{children}
					</Plaintext>
				);
			} else if (outline) {
				return (
					<Outline {...props}>
						{iconElement && <ButtonIcon>{iconElement}</ButtonIcon>}
						{children}
					</Outline>
				);
			} else if (underline) {
				return (
					<Underline {...props}>
						{iconElement && <ButtonIcon>{iconElement}</ButtonIcon>}
						{children}
					</Underline>
				);
			} else if (!getColoringType(props) && !props.color && !props.bg) {
				return (
					<DefaultButton {...props}>
						{iconElement && <ButtonIcon>{iconElement}</ButtonIcon>}
						{children}
					</DefaultButton>
				);
			} else {
				return (
					<Button {...props}>
						{iconElement && <ButtonIcon>{iconElement}</ButtonIcon>}
						{children}
					</Button>
				);
			}
		},
	),
) as React.ComponentClass<ButtonProps>;
