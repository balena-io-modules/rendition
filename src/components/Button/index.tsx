import {
	Button as GrommetButton,
	ButtonProps as GrommetButtonProps,
} from 'grommet';
import * as React from 'react';
import styled, { css } from 'styled-components';
import asRendition from '../../asRendition';
import {
	Coloring,
	RenditionSystemProps,
	ResponsiveStyle,
	Sizing,
	Theme,
} from '../../common-types';
import { useBreakpoint } from '../../hooks/useBreakpoint';
import { withConfirm, ConfirmOptions } from '../../internal/Confirm';
import {
	getColor,
	getColoringType,
	isLight,
	px,
	withConditional,
} from '../../utils';

const getHoverEffectOverride = (
	bg: string | undefined,
	color: string,
	opacity?: string,
) => {
	return css`
		&:hover:enabled,
		&:focus:enabled,
		&:active:enabled {
			svg {
				color: ${color} !important;
			}

			box-shadow: none;
			background: ${bg};
			border-color: ${bg};
			color: ${color};
			opacity: ${opacity ?? 'initial'};
		}
	`;
};

/* Using '&' to add an extra layer of specificity and make sure
the rendition styles override the Grommet global ones.
https://www.styled-components.com/docs/advanced#issues-with-specificity
*/
const ButtonBase = styled(GrommetButton)`
	& {
		font-weight: ${(props) => props.theme.button.font.weight};
		font-size: ${(props) => props.theme.button.font.size};
		height: ${(props) => props.theme.button.height};
		white-space: nowrap;
		svg {
			color: inherit !important;
			font-size: 0.925em;
		}
		/* These rules cause consistent styling when the button is rendered as a link */
		display: inline-flex;
		align-items: center;
		justify-content: center;

		&:disabled {
			cursor: not-allowed;
		}
	}
`;

const ColouredButton = styled(ButtonBase)`
	& {
		color: ${(props) =>
			isLight(getColor(props, 'color', 'main'))
				? props.theme.colors.text.main
				: 'white'};
		${(props) => {
			const color = getColor(props, 'color', 'dark');
			return getHoverEffectOverride(
				color,
				isLight(color) ? props.theme.colors.text.main : 'white',
			);
		}};
	}
`;

const Outline = styled(ButtonBase)`
	& {
		color: ${(props) => props.theme.colors.text.main};
		border-color: ${(props) =>
			getColor(props, 'color', 'main') || props.theme.colors.text.main};
		${(props) =>
			getHoverEffectOverride(
				props.color || props.theme.colors.text.main,
				isLight(getColor(props, 'color', 'main'))
					? props.theme.colors.text.main
					: 'white',
			)};
	}
`;

const Light = styled(ButtonBase)`
	& {
		color: ${(props) => props.theme.colors.text.main};
		border-color: ${(props) => props.theme.colors.text.main};
		background: white;
		border: none;
		${(props) =>
			getHoverEffectOverride(
				'white',
				props.theme.colors.secondary.main,
				'0.9',
			)};

		&:disabled {
			opacity: 1;
			color: #c6c8c9;
		}
	}
`;

const LightOutline = styled(Light)`
	& {
		color: white;
		border: 1px solid white;
		background: transparent;
		${(props) =>
			getHoverEffectOverride(
				'white',
				props.theme.colors.secondary.main,
				'0.9',
			)};

		&:disabled {
			opacity: 1;
			color: #c6c8c9;
		}
	}
`;

const Plain = styled(ButtonBase)`
	& {
		color: ${(props) =>
			getColor(props, 'color', 'main') || props.theme.colors.text.main};
		height: auto;
		font-weight: ${(props) => props.theme.button.font.weight};
		border-radius: 0;
		${(props) =>
			getHoverEffectOverride(
				'none',
				getColor(props, 'color', 'dark') || props.theme.colors.text.light,
			)};
	}
`;

const underlineButtonActiveStyles = (props: { theme: Theme }) => css`
	& {
		color: ${getColor(props, 'color', 'main') || props.theme.colors.text.main};
		background: none;
		box-shadow: 0px -1px 0 0px inset;
	}
`;

const Underline = styled(Plain)<{ active?: boolean; color?: string }>`
	& {
		${(props) => (props.active ? underlineButtonActiveStyles(props) : '')}
		border-bottom: 1px solid;
		background: none;
		&:hover:enabled,
		&:focus:enabled,
		&:active:enabled {
			${underlineButtonActiveStyles};
		}
	}
`;

const getStyledButton = (
	{
		outline,
		underline,
		plain,
		active,
		light,
	}: {
		outline?: boolean;
		underline?: boolean;
		plain?: boolean;
		active?: boolean;
		light?: boolean;
	},
	isPrimary: boolean,
) => {
	if (plain) {
		return Plain;
	}
	if (light) {
		if (outline) {
			return LightOutline;
		}
		return Light;
	}
	if (underline) {
		return Underline;
	}
	if ((outline && !active) || !isPrimary) {
		return Outline;
	}

	return ColouredButton;
};

const BaseButton = React.forwardRef((props: ThemedButtonProps, ref: any) => {
	const {
		outline,
		underline,
		children,
		label,
		primary,
		color,
		plain,
		active,
		light,
		compact,
		...restProps
	} = props;

	// Add a method to hide the button label. Here we can just pass an array of
	// boolean to decide on each breakpoint whether to show the label or not
	const shouldCompact = useBreakpoint(compact || [false]);

	// The 'primary' and 'color' props map to semantically different properties in
	// grommet, so mapping is performed here to keep the styling cleaner.
	// In grommet, 'primary' indicates that the button is filled with a solid
	// color, whereas in rendition 'primary' is a shorthand for the primary theme
	// color. In grommet 'color' refers to the fill or outline color of the
	// button, whereas in rendition it refers to the text color of the button
	let basePrimary =
		!!props.bg ||
		(!!getColoringType(props) && !outline && !underline && !props.plain);
	let baseColor = props.bg || getColor(props, 'bg', 'main');

	// Set default to 'secondary'
	baseColor = baseColor || props.theme.colors.secondary.main;

	// If the button is active, invert the background and text colors
	if (active) {
		basePrimary = !basePrimary;
	}

	const StyledButton = getStyledButton(
		{ underline, plain, active, light },
		basePrimary,
	);

	// Note that the 'plain' property is case to a Boolean to simplify Grommets
	// default behavior when providing an icon attribute and no label
	// see: https://github.com/grommet/grommet/issues/3030
	return (
		<StyledButton
			ref={ref}
			primary={basePrimary}
			color={baseColor}
			active={active && (underline || plain)}
			gap={px(props.theme.space[2])}
			{...restProps}
			plain={!!plain || !!underline}
			label={shouldCompact ? undefined : label || children}
		/>
	);
});

interface ButtonBaseProps extends Coloring, Sizing {
	width?: ResponsiveStyle;
	bg?: string;
	/** If true, the button will have a transparent background, and the border and text color will match */
	outline?: boolean;
	/** Similar to the plaintext prop, but displays a line underneath the button text */
	underline?: boolean;
	/** If true, use white background and default text color */
	light?: boolean;
	/** Optionally renders the label according to the value inside the array for each breakpoint */
	compact?: boolean[];
}

export interface InternalButtonProps
	extends ButtonBaseProps,
		Omit<React.HTMLAttributes<HTMLElement>, 'dir' | 'color' | 'children'>,
		GrommetButtonProps {
	type?: 'submit' | 'reset' | 'button';
	confirmation?: ConfirmOptions | string;
}

export type ButtonProps = InternalButtonProps & RenditionSystemProps;

export interface ThemedButtonProps extends ButtonProps {
	theme: Theme;
}
/** [View story source](https://github.com/balena-io-modules/rendition/blob/master/src/components/Button/Button.stories.tsx) */
export const Button = withConditional<ButtonProps>(withConfirm, (props) => {
	return 'confirmation' in props;
})(
	asRendition<React.ForwardRefExoticComponent<ButtonProps>>(
		BaseButton,
		[],
		['width', 'color', 'bg'],
	),
);
