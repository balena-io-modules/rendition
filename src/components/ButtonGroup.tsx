import styled from 'styled-components';
import { Flex } from './Grid';

export default styled(Flex)`
	> * {
		border-radius: 0;
		margin-right: -1px;

		&:hover {
			z-index: 1;
		}

		&:first-child {
			border-top-left-radius: ${props => props.theme.radius}px;
			border-bottom-left-radius: ${props => props.theme.radius}px;
		}

		&:last-child {
			border-top-right-radius: ${props => props.theme.radius}px;
			border-bottom-right-radius: ${props => props.theme.radius}px;
		}
	}
`;
