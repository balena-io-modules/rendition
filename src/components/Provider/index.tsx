import { Grommet } from 'grommet';
import merge from 'lodash/merge';
import * as React from 'react';
import styled from 'styled-components';
import { DefaultProps, Theme } from '../../common-types';
import defaultTheme from '../../theme';
import { px } from '../../utils';

const Base = styled(Grommet)`
	font-family: ${props => props.theme.font};
	font-size: ${props => px(props.theme.fontSizes[1])};

	h1,
	h2,
	h3,
	h4,
	h5,
	h6,
	button {
		font-family: ${props => props.theme.titleFont};
	}
`;

const Provider = ({ theme, ...props }: ThemedProvider) => {
	return <Base theme={merge(defaultTheme, theme)} {...props} />;
};

export interface ThemedProvider extends DefaultProps {
	theme?: Theme;
}

export default Provider;
