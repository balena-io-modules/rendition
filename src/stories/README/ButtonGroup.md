# ButtonGroup

Wrapper for buttons to make them stick together.

[View story source](https://github.com/balena-io-modules/rendition/blob/master/src/stories/ButtonGroup.js)