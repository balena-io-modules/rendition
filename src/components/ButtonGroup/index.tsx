import React from 'react';
import styled from 'styled-components';
import { Flex, FlexProps } from '../Flex';
import { Button } from '../Button';
import { TooltipProps } from '../../common-types';
import { px } from '../../utils';

export type OptionType = ButtonGroupOption | string | number | boolean;
export interface ButtonGroupOption {
	label: string | number;
	value: unknown;
	tooltip?: string | TooltipProps;
}
export interface ButtonGroupWithOptionsProps
	extends Omit<FlexProps, 'children'> {
	options: OptionType[];
	value?: OptionType;
	onGroupChange: (value: OptionType) => void;
}

export type ButtonGroupProps = FlexProps | ButtonGroupWithOptionsProps;

/**
 * Wrapper for buttons to make them stick together.
 *
 * [View story source](https://github.com/balena-io-modules/rendition/blob/master/src/components/ButtonGroup/ButtonGroup.stories.tsx)
 */
const ButtonGroupBase = styled(Flex)<{ isSelect?: boolean }>`
		${(props) =>
			props.isSelect
				? `
			border: 1px solid ${props.theme.global.control.border.color};
			width: fit-content;
			border-radius: ${props.theme.button.border.radius};
			padding: 2px;
			> button {
				padding: 2px 16px;
				border: 0px;
			}
		`
				: `
		> * {
			&:first-child {
				border-top-right-radius: 0;
				border-bottom-right-radius: 0;
				padding-left: ${px(props.theme.space[3])};
			}
	
			&:last-child {
				border-top-left-radius: 0;
				border-bottom-left-radius: 0;
				padding-right: ${px(props.theme.space[3])};
			}
	
			&:not(:last-child) {
				border-right: 0;
			}
	
			&:not(:last-child):not(:first-child) {
				border-radius: 0;
			}
	
			&:hover {
				z-index: 1;
			}
		`}
	}
`;

export const ButtonGroup = (props: ButtonGroupProps) => {
	if ('options' in props) {
		const { onGroupChange, height, ...otherProps } = props;
		return (
			<ButtonGroupBase isSelect {...otherProps} alignItems="center">
				{props.options.map((option) => {
					const label = typeof option !== 'object' ? option : option.label;
					return (
						<Button
							active={
								typeof option !== 'object'
									? option === props.value
									: option.value === props.value
							}
							tooltip={typeof option === 'object' ? option.tooltip : undefined}
							onClick={() => {
								if (onGroupChange) {
									onGroupChange(option);
								}
							}}
							height={height}
						>
							{label}
						</Button>
					);
				})}
			</ButtonGroupBase>
		);
	}

	if ('children' in props) {
		return <ButtonGroupBase {...props}>{props.children}</ButtonGroupBase>;
	}

	return null;
};
