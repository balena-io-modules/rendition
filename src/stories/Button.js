import * as React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { withScreenshot } from 'storybook-chrome-screenshot'
import withReadme from 'storybook-readme/with-readme'
import * as FaExpand from 'react-icons/lib/fa/expand'
import * as FaRecycle from 'react-icons/lib/fa/recycle'
import * as FaSpinner from 'react-icons/lib/fa/spinner'
import { Box, Button, Provider } from '../'
import * as Readme from './README/Button.md'

storiesOf('Core/Button', module)
  .addDecorator(withReadme(Readme))
  .addDecorator(withScreenshot())
  .add('Standard', () => {
    return (
      <Provider>
        <Box m={3}>
          <Button m={2} primary onClick={action('case-1')}>
            Primary
          </Button>
          <Button m={2} secondary onClick={action('case-2')}>
            Secondary
          </Button>
          <Button m={2} tertiary onClick={action('case-3')}>
            Tertiary
          </Button>
          <Button m={2} success onClick={action('case-4')}>
            Success
          </Button>
          <Button m={2} danger onClick={action('case-5')}>
            Danger
          </Button>
          <Button m={2} onClick={action('case-6')}>
            Default
          </Button>
        </Box>
      </Provider>
    )
  })
  .add('Emphasized', () => {
    return (
      <Provider>
        <Box m={3}>
          <Button m={2} emphasized primary onClick={action('case-1')}>
            Primary
          </Button>
          <Button m={2} emphasized secondary onClick={action('case-2')}>
            Secondary
          </Button>
          <Button m={2} emphasized tertiary onClick={action('case-3')}>
            Tertiary
          </Button>
          <Button m={2} emphasized success onClick={action('case-4')}>
            Success
          </Button>
          <Button m={2} emphasized danger onClick={action('case-5')}>
            Danger
          </Button>
          <Button m={2} emphasized onClick={action('case-6')}>
            Default
          </Button>
        </Box>
      </Provider>
    )
  })
  .add('Disabled', () => {
    return (
      <Provider>
        <Box m={3}>
          <Button m={2} disabled primary onClick={action('case-1')}>
            Primary
          </Button>
          <Button m={2} disabled secondary onClick={action('case-2')}>
            Secondary
          </Button>
          <Button m={2} disabled tertiary onClick={action('case-3')}>
            Tertiary
          </Button>
          <Button m={2} disabled success onClick={action('case-4')}>
            Success
          </Button>
          <Button m={2} disabled danger onClick={action('case-5')}>
            Danger
          </Button>
          <Button m={2} disabled onClick={action('case-6')}>
            Default
          </Button>
        </Box>
      </Provider>
    )
  })
  .add('Outline', () => {
    return (
      <Provider>
        <Box m={3}>
          <Button m={2} outline primary onClick={action('case-1')}>
            Primary
          </Button>
          <Button m={2} outline secondary onClick={action('case-2')}>
            Secondary
          </Button>
          <Button m={2} outline tertiary onClick={action('case-3')}>
            Tertiary
          </Button>
          <Button m={2} outline success onClick={action('case-4')}>
            Success
          </Button>
          <Button m={2} outline danger onClick={action('case-5')}>
            Danger
          </Button>
          <Button m={2} outline onClick={action('case-6')}>
            Default
          </Button>
        </Box>
      </Provider>
    )
  })
  .add('Underline', () => {
    return (
      <Provider>
        <Box m={3}>
          <Button m={2} primary underline onClick={action('case-1')}>
            Button
          </Button>
          <Button m={2} underline onClick={action('case-1')}>
            Button
          </Button>
        </Box>
      </Provider>
    )
  })
  .add('Plaintext', () => {
    return (
      <Provider>
        <Box m={3}>
          <Button m={2} plaintext primary onClick={action('case-1')}>
            Button
          </Button>
          <Button
            m={2}
            plaintext
            primary
            iconElement={<FaSpinner />}
            onClick={action('case-2')}
          >
            Button
          </Button>
        </Box>
      </Provider>
    )
  })
  .add('Width', () => {
    return (
      <Provider>
        <Box m={3}>
          <Button w={95} m={2} primary onClick={action('case-1')}>
            Button
          </Button>
          <Button w={60} m={2} secondary onClick={action('case-2')}>
            Button
          </Button>
          <Button w={120} m={2} tertiary onClick={action('case-3')}>
            Button
          </Button>
          <div style={{ width: 300 }}>
            <Button w='100%' m={2} onClick={action('case-4')}>
              Button
            </Button>
          </div>
        </Box>
      </Provider>
    )
  })
  .add('Square', () => {
    return (
      <Provider>
        <Box m={3}>
          <Button m={2} tertiary square onClick={action('case-3')}>
            <FaExpand />
          </Button>
          <Button m={2} square onClick={action('case-4')}>
            <FaRecycle />
          </Button>
        </Box>
      </Provider>
    )
  })
  .add('Square Emphasized', () => {
    return (
      <Provider>
        <Box m={3}>
          <Button m={2} emphasized tertiary square onClick={action('case-3')}>
            <FaExpand />
          </Button>
          <Button m={2} emphasized square onClick={action('case-4')}>
            <FaRecycle />
          </Button>
        </Box>
      </Provider>
    )
  })
