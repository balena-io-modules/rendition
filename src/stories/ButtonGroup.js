import * as React from 'react'
import { storiesOf } from '@storybook/react'
import { withScreenshot } from 'storybook-chrome-screenshot'
import withReadme from 'storybook-readme/with-readme'
import { Box, Button, ButtonGroup, Provider } from '../'
import * as Readme from './README/ButtonGroup.md'

storiesOf('Core/ButtonGroup', module)
  .addDecorator(withReadme(Readme))
  .addDecorator(withScreenshot())
  .add('Standard', () => {
    return (
      <Provider>
        <Box m={3}>
          <ButtonGroup>
            <Button>First</Button>
            <Button>Second</Button>
            <Button>Third</Button>
          </ButtonGroup>
        </Box>
      </Provider>
    )
  })
