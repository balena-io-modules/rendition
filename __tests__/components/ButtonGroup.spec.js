/* globals expect, test */
import React from 'react'
import renderer from 'react-test-renderer'
import Provider from '../../src/components/Provider'
import Button from '../../src/components/Button'
import ButtonGroup from '../../src/components/ButtonGroup'

test('ButtonGroup renders correctly', () => {
  const component = renderer.create(
    <Provider>
      <ButtonGroup>
        <Button>First</Button>
        <Button>Second</Button>
        <Button>Third</Button>
      </ButtonGroup>
    </Provider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
