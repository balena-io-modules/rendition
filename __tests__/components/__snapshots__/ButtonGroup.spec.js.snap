// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup renders correctly 1`] = `
<div
  className="sc-bdVaJa jKopLr"
>
  <div
    className="sc-kgoBCf hzrgZk sc-kAzzGY dirfrO sc-jzJRlG cmnKTT sc-fjdhpX jmbwCw"
  >
    <button
      className="sc-VigVT hegYug sc-gqjmRU gECYIb"
    >
      First
    </button>
    <button
      className="sc-VigVT hegYug sc-gqjmRU gECYIb"
    >
      Second
    </button>
    <button
      className="sc-VigVT hegYug sc-gqjmRU gECYIb"
    >
      Third
    </button>
  </div>
</div>
`;
